numpy/random/mtrand.pyx

```diff
@@ -3737,6 +3782,16 @@ cdef class RandomState:
         final_shape.append(mean.shape[0])
         x = self.standard_normal(final_shape).reshape(-1, mean.shape[0])
 
+        if mode == 'factor':
+            factor = cov
+            # check if factor is a upper-triangular matrix (e.g Cholesky factor)
+            if np.allclose(factor, np.triu(factor)):
```

Comment:
If the goal is performance perhaps this check could also be done only in the case `check_valid` is True?

Reply:
Also you can simply check the lower part to be zero and don't bother with the upper part. But this should be mentioned that now it only works with upper triangular chol factors.